src/tir/schedule/primitive/block_annotate.cc

```diff
@@ -0,0 +1,313 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+#include "../utils.h"
+
+namespace tvm {
+namespace tir {
+
+class StorageAlignAxisOutOfRangeError : public ScheduleError {
+ public:
+  explicit StorageAlignAxisOutOfRangeError(IRModule mod, Buffer buffer, int axis)
+      : mod_(std::move(mod)), buffer_(std::move(buffer)), axis_(axis) {}
+
+  String FastErrorString() const final {
+    return "ScheduleError: The input `axis` is out of range. It is required to be in range "
+           "[-ndim, ndim) where `ndim` is the number of dimensions of the buffer to set "
+           "storage alignment.";
+  }
+
+  String DetailRenderTemplate() const final {
+    std::ostringstream os;
+    int ndim = static_cast<int>(buffer_->shape.size());
+    os << "The buffer to set storage alignment " << buffer_->name << " has " << ndim
+       << " dimension(s), so `axis` is required to be in [" << -(ndim) << ", " << ndim
+       << ") for storage_align. However, the input `axis` is " << axis_
+       << ", which is out of the expected range.";
+    return os.str();
+  }
+
+  IRModule mod() const final { return mod_; }
+  Array<ObjectRef> LocationsOfInterest() const final { return {}; }
```

Comment:
Good point, @junrushao1994 do we have convention what should be put in location of interests?

Reply:
Yes, we only put "components of IR", that means, blocks and loops, in the location of interests.

Reply:
Yeah that's correct

Reply:
Could we add where buffers are declared to locations of interest? Then we can point to them. But it might not be helpful if it always point to the function inputs.